_docs/rfc/scaffolding_new_modules.md

```diff
@@ -0,0 +1,402 @@
+  not intuitive (E.g whitespace chomping).
+- Requires some work to embed into `terragrunt` so that arbitrary templates can be invoked.
+- We need to invent a DSL for handling and declaring template inputs.
+
+### Template tests
+
+Template tests can be invoked using the `shell` module in `terratest` that can be used to call `terragrunt new` using
+these templates with arbitrary input values. You can then apply the generated template to test the generated code
+against various variable inputs.
+
+### Template structure
+
+_TODO: This still needs to be fleshed out._
+
+
+### FAQ
+
+#### When should code be generated?
+
+The challenge with code generation is deciding when, and how often to generate the code. For example, suppose you ran
+`terragrunt new component` and it seeded the live configuration with all the different variables of the underlying
+module. Next, you will need to make the changes to set the values for each of the variables. This makes sense to edit
+directly on the generated code, so you shouldn't have to run the generator again. However, suppose that at a later point
+the module is updated and a new variable is introduced. Should you regenerate the code so you get the updated variable
+description?
+
+While "codegen drift" is a concern, I think the only practical usage of the proposed scaffolding is for bootstrapping an
+initial implementation, and not for maintaining the code through code generation. As such, you should only generate the
+code in the initial implementation, and then proceed to make manual edits to the generated code over time to maintain
+it, rather than going through `terragrunt new` each time to roll out an update.
+
+
+#### Should the generated code be checked in?
+
+Given the answer to [When should code be generated?](#when-should-code-be-generated), yes it should be checked in so
+that you can maintain the edits being made to it.
+
+
+#### How to keep code in sync with template?
+
+In the scenario covered in [When should code be generated?](#when-should-code-be-generated) there is a question of how
+to introduce the new variable to a previously generated code. Specifically, when you introduce a new variable, and if
+you were keeping the list of variable inputs in the live code that was generated, you could end up in a situation of
+"codegen drift", where the module has a new variable that is omitted from your live config.
+
+Unfortunately, as mentioned in the answer to [When should code be generated?](#when-should-code-be-generated), it is not
+practical to always regenerate the code from the template every time the module changes. This means that we are once
+again back to relying on developer discipline for maintaining this link.
+
+The solution proposed here will not address this challenge. However, the hope is that the proposed solution will improve
+the UX over what we have currently, while acknowledging that it is not a perfect solution that solves all challenges.
```

Comment:
One potential thing we could do is add a `terragrunt new input-variable` command that adds the variable to both an existing modules in infra-modules and a `terragrunt.hcl` file (or files) in infra-live... But building scaffolding to generate code into existing files is a bit tricky.